web/src/layouts/PublicLayout.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from 'react'
+import { Link, Outlet, useNavigate } from 'react-router-dom'
+import RegisterModal from '../components/RegisterModal'
+import LoginModal from '../components/LoginModal'
+
+export default function PublicLayout() {
```

Comment:
navbar markup with the logo and buttons is 80% the same as logged-in version in AppLayout, need to make it DRY. again, it repeats itself